### PR TITLE
Add .codeclimate.yml file

### LINKED ISSUE #42: Add a .codeclimate.yml file to this repo
It is possible to configure how CodeClimate looks at a repository.
[These Docs](https://docs.codeclimate.com/docs/getting-started-configuration) explain how this is done. Please create an appropriate .codeclimate.yml for this repo. It should test the main language of the repo and run tests.

### PR DESCRIPTION
**Pull request by rzats**
Fixes #42

This is a standard (ie autogenerated) file that runs [Checkstyle](https://docs.codeclimate.com/v1.0/docs/checkstyle), [PMD](https://docs.codeclimate.com/v1.0/docs/pmd) and [fixme](https://docs.codeclimate.com/v1.0/docs/fixme) on all Java source files. Let me know if any of this needs to be tweaked :)

### CODE REVIEW

**Comment by SendGridDX** [comment]
[CLA assistant check](https://cla.sendgrid.com/sendgrid/smtpapi-java?pullRequest=43) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.sendgrid.com/sendgrid/smtpapi-java?pullRequest=43) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.sendgrid.com/check/sendgrid/smtpapi-java?pullRequest=43) it.</sub>

File: .codeclimate.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+---
+engines:
+  checkstyle:
+    enabled: true
+  duplication:
+    enabled: true
+    config:
+      languages:
+      - php
```
Comment (mbernier) [inline .codeclimate.yml]: Can you please modify this for a java code base? :) 

Note: we added the same issue on our 3 PHP repos as well (I am not sure as of right now if those issues are still available)

File: .codeclimate.yml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+---
+engines:
```
Comment (mbernier) [inline .codeclimate.yml]:
```
codeclimate validate-config
WARNING: A new version (v0.68.0) is available. Upgrade instructions are available at: https://github.com/codeclimate/codeclimate#packages
WARNING: 'engines' has been deprecated, please use 'plugins' instead
WARNING: 'ratings' has been deprecated, and will not be used
```
Can you please remove ratings and change engines to plugins?

**Comment by thinkingserious** [comment]
Hello @rzats,<br /><br/>Thanks again for the PR!<br />                                              <br />We want to show our appreciation by sending                                              you some swag. Could you please fill out                                              <a href="https://dx.sendgrid.com/swag">this form</a> so we can send it to you? Thanks!                                             <br /><br />Team SendGrid DX

**Pull request merged**